installer/environment.py

```
@@ -70,77 +109,94 @@
     def get_version(self):
         version_file = self.project_root / "version.txt"
         if version_file.exists():
-            with open(version_file, 'r') as f:
-                return f.read().strip()
+            try:
+                with open(version_file, 'r') as f:
+                    return f.read().strip()
+            except IOError as e:
+                raise Exception(f"File read error: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)